src/pyodide-py/pyodide/_state.py

```diff
@@ -0,0 +1,40 @@
+import __main__
+import sys
+import gc
+
+from ._core import JsProxy
+from ._importhooks import jsfinder
+
+
+def save_state():
+    loaded_js_modules = {}
+    for [key, value] in sys.modules.items():
+        if isinstance(value, JsProxy):
+            loaded_js_modules[key] = value
+
+    return dict(
+        globals=dict(__main__.__dict__),
+        js_modules=dict(jsfinder.jsproxies),
+        loaded_js_modules=loaded_js_modules,
+    )
+
+
+def restore_state(state):
```

Comment:
Please add type annotations and at least a 1 line summary docstring if they are private.